Utility helpers to convert between std::vector and NSArray.

shoumikhin opened the pull request:
Summary: .

Differential Revision: D71752746

shoumikhin commented:
@pytorchbot label "topic: not user facing"

swolchok commented on extension/apple/ExecuTorch/Internal/ExecuTorchUtils.h:
```diff
@@ -93,6 +93,49 @@ static inline T extractValue(NSNumber *number) {
+ * @return A std::vector with the values extracted as type T.
+ */
+template <typename T>
+static inline std::vector<T> toVector(NSArray<NSNumber *> *array) {
```
doesn't need to be `static` or `inline`

shoumikhin replied:
Thanks! Removed.

swolchok commented on extension/apple/ExecuTorch/Internal/ExecuTorchUtils.h:
```diff
@@ -93,6 +93,49 @@ static inline T extractValue(NSNumber *number) {
+ * @return An NSArray populated with NSNumber objects representing the container's items.
+ */
+template <typename Container>
+static inline NSArray<NSNumber *> *toNSArray(const Container &container) {
```
ditto

swolchok reviewed (approved):
it's inefficient to need these, but when you do need them (and I'm totally prepared to believe you do) then yep they look good.

shoumikhin committed:
Utility helpers to convert between std::vector and NSArray. (#9597)

Summary:

.

Reviewed By: swolchok, kirklandsign

Differential Revision: D71752746
